alacritty_terminal/tests/ref/underline/alacritty.recording

```diff
@@ -1,13 +1,15 @@
-[undeadleech@undeadlap underline]$ 
[K[undeadleech@undeadlap underline]$ echo -e "\e[4mUNDERLINED\e[0m"
-[4mUNDERLINED[0m
-[undeadleech@undeadlap underline]$ echo -e "\e[4:1mUNDERLINED\e[4:0m"
-[4:1mUNDERLINED[4:0m
-[undeadleech@undeadlap underline]$ echo -e "\e[4:2mUNDERLINED\e[24m"
-[4:2mUNDERLINED[24m
-[undeadleech@undeadlap underline]$ echo -e "\e[4:3mUNDERLINED\e[21m";mUNDERLINED\e[21m"2mUNDERLINED\e[21m"1mUNDERLINED\e[21m"[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[1Pm"[1Pm"0m"
-[4:3;21mUNDERLINED[0m
-[undeadleech@undeadlap underline]$ echo -e "\e[4;4:2mUNDERLINED\e[0m"
-[4;4:2mUNDERLINED[0m
-[undeadleech@undeadlap underline]$ echo -e "\e[4;4:2mUNDERLINED\e[0m":;4:2mUNDERLINED\e[0m"1;4:2mUNDERLINED\e[0m"[C[C[C[1P;4:2mUNDERLINED\e[0m"2;4:2mUNDERLINED\e[0m"[C[C[C[C[1PmUNDERLINED\e[0m"1mUNDERLINED\e[0m"
-[4:2;4:1mUNDERLINED[0m
-[undeadleech@undeadlap underline]$ 
\ No newline at end of file
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ 
[K$ [7mecho -e "\e[4mUNDERLINED\e[0m"[27m
[C[Cecho -e "\e[4mUNDERLINED\e[0m"
+[?2004l
[4mUNDERLINED[0m
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ [7mecho -e "\e[4:1mUNDERLINED\e[4:0m"[27m
[C[Cecho -e "\e[4:1mUNDERLINED\e[4:0m"
+[?2004l
[4:1mUNDERLINED[4:0m
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ [7mecho -e "\e[4:2mUNDERLINED\e[24m"[27m
[C[Cecho -e "\e[4:2mUNDERLINED\e[24m"
+[?2004l
[4:2mUNDERLINED[24m
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ [7mecho -e "\e[4:3mUNDERLINED\e[0m"[27m
[C[Cecho -e "\e[4:3mUNDERLINED\e[0m"
+[?2004l
[4:3mUNDERLINED[0m
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ [7mecho -e "\e[4;4:2mUNDERLINED\e[0m"[27m
[C[Cecho -e "\e[4;4:2mUNDERLINED\e[0m"
+[?2004l
[4;4:2mUNDERLINED[0m
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ [7mecho -e "\e[4:2;4:1mUNDERLINED\e[0m"[27m
[C[Cecho -e "\e[4:2;4:1mUNDERLINED\e[0m"
+[?2004l
[4:2;4:1mUNDERLINED[0m
+]0;jesse@pohjala:~/dev/oss/alacritty[?2004h$ [7mecho -e "\e[58;5;2;38;5;3mNO\e[4mUND\e[58;5;6mERLI\e[59mNE\e[0m"[27m
[C[Cecho -e "\e[58;5;2;38;5;3mNO\e[4mUND\e[58;5;6mERLI\e[59mNE\e[0m"
+[?2004l
[58;5;2;38;5;3mNO[4mUND[58;5;6mERLI[59mNE[0m

```

Comment:
You only ever use the semicolon separated form here, however that's technically the legacy version and the preferred one should be the colon separated one.

So at the very least we should have a test that verifies both the `5` and `2` forms work properly with the colon separated one.